### PR TITLE
Add cloud_sql_target field in google_data_loss_prevention_discovery_config

### PR DESCRIPTION
**Pull request by patrickmoy**
**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
dlp: added `cloud_sql_target` field to `google_data_loss_prevention_discovery_config` resource
```

### CODE REVIEW

File: mmv1/third_party/terraform/services/datalossprevention/resource_data_loss_prevention_discovery_config_test.go [hunk 1]
```diff
@@ -584,3 +620,121 @@ resource "google_data_loss_prevention_discovery_config" "basic" {
...
+            }
+            disabled {}
+        }
+    }
+    targets {
+        cloud_sql_target {
+            filter {
+                others {}
+            }
+			generation_cadence {
```
Comment (slevenick) [inline mmv1/third_party/terraform/services/datalossprevention/resource_data_loss_prevention_discovery_config_test.go:666]: Please fix the spacing in this file, may be a tabs vs spaces issue

File: mmv1/third_party/terraform/services/datalossprevention/resource_data_loss_prevention_discovery_config_test.go [hunk 2]
```diff
@@ -584,3 +620,121 @@ resource "google_data_loss_prevention_discovery_config" "basic" {
...
+                        patterns {
+                            project_id_regex = ".*"
+                            instance_regex = ".*"
+                            database_regex = "do-not-scan.*"
+                            database_resource_name_regex = ".*"
+                        }
+                    }
+                }
+            }
+			conditions {
```
Comment (slevenick) [inline mmv1/third_party/terraform/services/datalossprevention/resource_data_loss_prevention_discovery_config_test.go:654]: Spacing

File: mmv1/third_party/terraform/services/datalossprevention/resource_data_loss_prevention_discovery_config_test.go [hunk 3]
```diff
@@ -584,3 +620,121 @@ resource "google_data_loss_prevention_discovery_config" "basic" {
...
+	inspect_config {
+		info_types {
+			name = "EMAIL_ADDRESS"
+		}
+	}
+}
+resource "google_data_loss_prevention_discovery_config" "basic" {
+	parent = "projects/%{project}/locations/%{location}"
+	location = "%{location}"
+	status = "RUNNING"
```
Comment (slevenick) [inline mmv1/third_party/terraform/services/datalossprevention/resource_data_loss_prevention_discovery_config_test.go:639]: Spacing

File: mmv1/third_party/terraform/services/datalossprevention/resource_data_loss_prevention_discovery_config_test.go [hunk 4]
```diff
@@ -584,3 +620,121 @@ resource "google_data_loss_prevention_discovery_config" "basic" {
...
+                schema_modified_cadence {
+                    types = ["NEW_COLUMNS", "REMOVED_COLUMNS"]
+                    frequency = "UPDATE_FREQUENCY_DAILY"
+                }
+                refresh_frequency = "UPDATE_FREQUENCY_MONTHLY"
+            }
+        }
+	}
+	targets {
+        cloud_sql_target {
```
Comment (slevenick) [inline mmv1/third_party/terraform/services/datalossprevention/resource_data_loss_prevention_discovery_config_test.go:724]: Spacing

**Review by slevenick — CHANGES_REQUESTED** [review]
Overall looks good, just some minor notes. Thanks for adding an update test for this!

File: mmv1/products/dlp/DiscoveryConfig.yaml [hunk 5]
```diff
@@ -325,6 +330,114 @@ properties:
                 # The fields below are necessary to include the "disabled" filter in the payload
               send_empty_value: true
               allow_empty_object: true
+        - !ruby/object:Api::Type::NestedObject
+          name: cloudSqlTarget
+          description: 'Cloud SQL target for Discovery. The first target to match a table will be the one applied.'
+          properties:
+            - !ruby/object:Api::Type::NestedObject
+              name: filter
+              description: 'Required. The tables the discovery cadence applies to. The first target with a matching filter will be the one to apply to a table.'
```
[the remark this answers is not in the document]
Comment (patrickmoy) [reply mmv1/products/dlp/DiscoveryConfig.yaml:339]: Done.

File: mmv1/third_party/terraform/services/datalossprevention/resource_data_loss_prevention_discovery_config_test.go (hunk 2 above)
Comment (patrickmoy) [reply mmv1/third_party/terraform/services/datalossprevention/resource_data_loss_prevention_discovery_config_test.go:654]: Done.

File: mmv1/third_party/terraform/services/datalossprevention/resource_data_loss_prevention_discovery_config_test.go (hunk 3 above)
Comment (patrickmoy) [reply mmv1/third_party/terraform/services/datalossprevention/resource_data_loss_prevention_discovery_config_test.go:639]: Done.

File: mmv1/third_party/terraform/services/datalossprevention/resource_data_loss_prevention_discovery_config_test.go (hunk 4 above)
Comment (patrickmoy) [reply mmv1/third_party/terraform/services/datalossprevention/resource_data_loss_prevention_discovery_config_test.go:724]: Done.

**Comment by patrickmoy** [comment]
Made indentation changes, added required tag, and squashed fixes into 1 commit (total = 2: 1 base commit to add support, 1 commit with feedback fixes)

**Review by slevenick — APPROVED** [review]

**Pull request merged**